### ISSUE TITLE
Missing dependency in buildbot[bundle] for Python3 support

### ISSUE THREAD

**Issue by perror** [buildbot/buildbot#3495]
I recently tried to run buildbot from a fresh `pip install`, it worked fine in Python2 but the installation procedure failed when running it under a Python3 virtual environment.

My problem was quite simple to solve, it just required to install the `wheel` package to get through the installation procedure. Maybe you should consider adding the dependency to your list of packages (just in case the user do not have it on his own system just like me).

**Comment by rodrigc** [comment]
How did you create the virtual environments in Python 2 and Python 3?
What versions of pip and virtualenv are on your system?

**Comment by tardyp** [comment]
normally wheel is not required to install buildbot. Only to build it. reasonably recent pip should be able to install wheels without requiring wheel package

**Comment by perror** [comment]
I am on a Debian 'unstable' (updated daily), and this problem occurs only on Python3-side, I ran through a complete installation and run of buildbot under Python2 without any problem. 

Here are the commands I type:

    $> pyvenv buildbot-sandbox
    $> cd buildbot-sandbox/
    $> . bin/activate
    (buildbot-sandbox) $> pip install -U pip
    Requirement already up-to-date: pip in ./lib/python3.5/site-packages
    (buildbot-sandbox) $> pip install 'buildbot[bundle]'

I get the packages downloaded properly, but then most of the packages are failing like this (here the `future` package):

    Building wheels for collected packages: future, Twisted, sqlalchemy, sqlalchemy-migrate, MarkupSafe, Tempita
      Running setup.py bdist_wheel for future ... error
      Complete output from command /home/fleury/tmp/Buildbot/buildbot-sandbox/bin/python3.5 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-7r4e31_r/future/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/tmpdhk60rxdpip-wheel- --python-tag cp35:
      usage: -c [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
         or: -c --help [cmd1 cmd2 ...]
         or: -c --help-commands
         or: -c cmd --help
  
      error: invalid command 'bdist_wheel'
  
      ----------------------------------------
      Failed building wheel for future

Then, when I redo the process but I take care of installing `wheel` before installing buildbot, everything is fine. Note, I simply install the `wheel` package like this:

    (buildbot-sandbox) $> pip install wheel

**Comment by tardyp** [comment]
> Building wheels for collected packages: future, Twisted, sqlalchemy, sqlalchemy-migrate, MarkupSafe, Tempita

I think this is a problem in pip. Here pip is trying to build wheel cache to speedup further installs. It shouldn't if wheel package is not present.

**Comment by perror** [comment]
I am not really familiar with `pip`, so I am not really understanding all the details. If this is correct, then you can obviously close the issue.

**Comment by tardyp** [comment]
issue raised on pypa/pip ...

**Comment by rodrigc** [comment]
I ran into the same issue when trying to set up buildbot in a Debian 9.1 Stretch VM
running on DigitalOcean.  I had to `apt install python3-wheel` before creating the virtual environments.  It was a bit hard to follow the source of the error.

**Comment by tardyp** [comment]
@rodrigc please discuss it in pip bug. This is something that they need to work with the distros.

**Comment by p12tic** [comment]
I think this bug is no longer present, especially since we've completed the 2->3 python migration. Please reopen this issue if this still happens.
